Sadot/User/OrderManagement/OrderForm.cs

```diff
@@ -1239,16 +1242,32 @@ private void btnPayPartOfBill_Click_1(object sender, EventArgs e)
         private bool CheckIfPartOfBillAllreadyPaid(int id)
         {
             bool flag = false;
-            for(int i = 0; i < linesInExistOrders.Count; i++)
+            for (int i = 0; i < linesInExistOrders.Count; i++)
             {
                 if (linesInExistOrders[i].ProductID == id)
                 {
                     lineInOrder = linesInExistOrders[i];
                     flag = true;
                 }
-                    
+
             }
             return flag;
         }
+
+        /// <summary>
+        /// method wich works when the user press on "MarkOrderAsReceived" button
+        /// the method will update current table OrderState to "order received"
+        /// </summary>
+        private void btnMarkOrderAsReceived_Click(object sender, EventArgs e)
+        {
+            if (table.OrderState == "הזמנה בהכנה")
+            {
+                table.TimeOfOrder = DateTime.Now - table.TimeOfOrder.TimeOfDay;
+                table.OrderState = "הזמנה התקבלה";
```

Comment:
You can unite 1267,1268 - you are updating these together for every table

Reply:
Fixed